### ISSUE TITLE
CW generates invalid URNs for some resource types

### ISSUE THREAD

**Issue by fwojciec** [CloudWanderer-io/CloudWanderer#260]
Happy New Year Sam!

We've found two edge cases where incorrect URNs are generated for specific (dependent) resource types:

- autoscaling -> auto_scaling_group: ID parts is a nested list, for example:
```python
[['Caching Proxy ASG'], ['some_id'], ['openapi-proxy-autoscaling']]
```
- lambda -> layer: ID parts includes an `int` (layer version)

I can look into making a PR later, but would have to read the code to figure out where in CW exactly the ID generation is taking place... For now we're handling it in our app when parsing the CW URN, but I doubt this is intended behavior so probably needs fixing.

**Comment by Sam-Martin** [comment]
Interesting! I can't reproduce the autoscaling group one...
```
from cloudwanderer.aws_interface import CloudWandererAWSInterface


def test_autoscaling_auto_scaling_group():
    interface = CloudWandererAWSInterface()

    result = next(
        interface.get_resources(service_name="autoscaling", resource_type="auto_scaling_group", region="eu-west-1")
    )

    assert result.urn == []
```
outputs:
```
 AssertionError: assert URN(cloud_name='aws', account_id='080863329876', region='eu-west-1', service='autoscaling', resource_type='auto_scaling_group', resource_id_parts=['test']) == []
```
Which doesn't show the nested list? I'm confused as to how auto scaling group is getting multiple id parts in the first place for you.
Can you provide some example (anonymised) `DescribeAutoScalingGroups` output?

I can however replicate the issue you're talking about with layers.
```
from cloudwanderer.aws_interface import CloudWandererAWSInterface


def test_autoscaling_auto_scaling_group():
    interface = CloudWandererAWSInterface()

    result = next(interface.get_resources(service_name="lambda", resource_type="layer", region="eu-west-1"))

    assert result.urn == []
```
output:
```
URN(cloud_name='aws', account_id='080863329876', region='eu-west-1', service='lambda', resource_type='layer_version', resource_id_parts=['cwsyn-talkvegantome-0284f868-8a5f-42de-a7c0-806ae4b55044', 2])
```
which shows the version as an integer.

I _think_ this behaviour was preserved originally due to boto3's [`get_layer_version` argument requiring an integer](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html#Lambda.Client.get_layer_version) but I agree it needs fixing somehow as it is problematic as URNs should be wholly stringy!

I'll see what I can do with the lambda layers integer problem, but let me know if you find any more info about the ASG issue as without being able to reproduce it I'm at a bit of a loss!

**Comment by Sam-Martin** [comment]
Integer and list `resource_id_parts` should no longer be possible as CloudWanderer `URN` now autoconverts `int` to `str` and throws a `ValueError` if anything else which is not a string is passed in.

Try updating to 0.29.1 and raise a new issue if you still find nested lists occurring.

**Comment by fwojciec** [comment]
Thank you!

**Comment by Sam-Martin** [comment]
Thanks @0bart ! Fixed in 0.29.2 with https://github.com/CloudWanderer-io/CloudWanderer/pull/265
